### PR TITLE
test/fuzz: add crypto/merkle.FuzzProofsFromByteSlices

### PR DESCRIPTION
**Pull request by odeke-em**
Adds a fuzzer to exorcise merkle.ProofsFromByteSlices.

### CODE REVIEW

File: test/fuzz/tests/crypto_merkle_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+package tests
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/tendermint/tendermint/crypto/merkle"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
```
Comment (creachadair) [inline test/fuzz/tests/crypto_merkle_test.go:11]:
```suggestion
	tmcrypto "github.com/tendermint/tendermint/crypto"
	"github.com/tendermint/tendermint/crypto/merkle"
	tmrand "github.com/tendermint/tendermint/libs/rand"
```

File: test/fuzz/tests/crypto_merkle_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+func FuzzProofsFromByteSlices(f *testing.F) {
+	seeds := [][]byte{
+		nil,
+		[]byte{},
```
Comment (creachadair) [inline test/fuzz/tests/crypto_merkle_test.go:17]:
```suggestion
		{},
```

**Comment by odeke-em** [comment]
@creachadair @marbar3778 could y'all please help me reopen this PR? I was on vacation and github-actions bot closed this PR. Thank you.

File: test/fuzz/tests/crypto_merkle_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	// [][]byte isn't supported as an input hence add []byte values.
+	var fakeSeparator = []byte("***FAKE_MARKER***")
+
+	for i := 0; i < 10; i++ {
+		total := 100
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply test/fuzz/tests/crypto_merkle_test.go:24]: This is kind of pedantic :-)

**Pull request closed without merging**